### PR TITLE
Update API routes for remote_db id and secret id

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Update Core and Front tables routes to take into account `remote_database_table_id` and `remote_database_secret_id`.

Changes in the public API: 
Post of a table takes optional params `remote_database_table_id` and `remote_database_secret_id`.
Somehow this is not taken by swagger. 

We were returning different types on the routes to list the tables or get a table. I'm now returning the same type for both, stored in `CoreAPITablePublic` for more consistency. This means the 3 fields `timestamp`, `tags` & `parents` are now retrieved on the list route. 

## Risk

Break those routes. 
Locally tested: upload a table, query it from an assistant, delete it. 

## Deploy Plan

Deploy Core. 
Deploy front. 
Deploy Docs.

### CODE REVIEW

File: front/public/swagger.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
       "get": {
         "summary": "Get an app run",
         "description": "Retrieve a run for an app in the workspace identified by {wId}.",
-        "tags": ["Apps"],
```
Comment (PopDaph) [inline front/public/swagger.json:18]: All these changes were generated because I launched `npm run docs` locally.

Comment (PopDaph) [reply front/public/swagger.json]: OK so basically I ran `npm run docs` which generated a lot of changes on this file, but this does not pass the linter, so I linted it, and it put it exactly back as it was before. 😄

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/index.ts [hunk 2]
```diff
@@ -14,30 +14,26 @@ import logger from "@app/logger/logger";
...
 
-const UpsertDatabaseTableRequestBodySchema = t.type({
-  table_id: t.union([t.string, t.undefined]),
-  name: t.string,
-  description: t.string,
-  timestamp: t.union([t.number, t.undefined, t.null]),
-  tags: t.union([t.array(t.string), t.undefined, t.null]),
-  parents: t.union([t.array(t.string), t.undefined, t.null]),
-});
+const UpsertDatabaseTableRequestBodySchema = t.intersection([
```
Comment (PopDaph) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/index.ts:20]: cc @albandum looks like this was not taken by npm run docs: I added two optional new params to the route to post a table.

File: types/src/front/lib/core_api.ts [hunk 3]
```diff
@@ -128,6 +126,13 @@ export type CoreAPITable = {
   parents: string[];
 };
 
+export type CoreAPITable = CoreAPITablePublic & {
```
Comment (Fraggle) [inline types/src/front/lib/core_api.ts:129]: I notice that we don't suffix with "Type" here (contrary to other types I have seen so far), any reason why that I should be aware of ?

**Review by Fraggle — APPROVED** [review]
LGTM

File: types/src/front/lib/core_api.ts (hunk 3 above)
Comment (fontanierh) [reply types/src/front/lib/core_api.ts:129]: I think we are suffixing with `Type` due to name-clashing with the models. `User` is the sequelize model, `UserType`  is the data-type that we typically return from `front`'s API

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/index.ts (hunk 2 above)
Comment (albandum) [reply front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/index.ts:20]: You need to update the top jsdoc above the endpoint for the docs to pick it up! I'll push a small commit to your PR to update

**Review by fontanierh — APPROVED** [review]
LGTM

File: types/src/front/lib/core_api.ts (hunk 3 above)
Comment (PopDaph) [reply types/src/front/lib/core_api.ts:129]: I have the same rule in mind indeed 👍🏻

**Pull request merged**